schema/pgsql/schema.sql

```diff
@@ -112,51 +120,60 @@ CREATE TABLE rotation (
     -- Used by Web to avoid showing shifts that never happened
     actual_handoff bigint NOT NULL,
 
+    CONSTRAINT pk_rotation PRIMARY KEY (id),
+
     -- each schedule can only have one rotation with a given priority starting at a given date
-    UNIQUE (schedule_id, priority, first_handoff),
+    CONSTRAINT key_rotation_schedule_schedule_priority_first_handoff UNIQUE (schedule_id, priority, first_handoff),
 
-    CONSTRAINT pk_rotation PRIMARY KEY (id)
+    CONSTRAINT fk_rotation_schedule FOREIGN KEY (schedule_id) REFERENCES schedule(id)
 );
 
 CREATE TABLE timeperiod (
     id bigserial,
-    owned_by_rotation_id bigint REFERENCES rotation(id), -- nullable for future standalone timeperiods
+    owned_by_rotation_id bigint, -- nullable for future standalone timeperiods
 
-    CONSTRAINT pk_timeperiod PRIMARY KEY (id)
+    CONSTRAINT pk_timeperiod PRIMARY KEY (id),
+    CONSTRAINT fk_timeperiod_rotation FOREIGN KEY (owned_by_rotation_id) REFERENCES rotation(id)
 );
 
 CREATE TABLE rotation_member (
     id bigserial,
-    rotation_id bigint NOT NULL REFERENCES rotation(id),
-    contact_id bigint REFERENCES contact(id),
-    contactgroup_id bigint REFERENCES contactgroup(id),
+    rotation_id bigint NOT NULL,
+    contact_id bigint,
+    contactgroup_id bigint,
     position integer NOT NULL,
 
-    UNIQUE (rotation_id, position), -- each position in a rotation can only be used once
+    -- each position in a rotation can only be used once
+    CONSTRAINT key_rotation_member_rotation_position UNIQUE (rotation_id, position),
 
     -- Two UNIQUE constraints prevent duplicate memberships of the same contact or contactgroup in a single rotation.
     -- Multiple NULLs are not considered to be duplicates, so rows with a contact_id but no contactgroup_id are
     -- basically ignored in the UNIQUE constraint over contactgroup_id and vice versa. The CHECK constraint below
     -- ensures that each row has only non-NULL values in one of these constraints.
-    UNIQUE (rotation_id, contact_id),
-    UNIQUE (rotation_id, contactgroup_id),
-    CHECK (num_nonnulls(contact_id, contactgroup_id) = 1),
-
-    CONSTRAINT pk_rotation_member PRIMARY KEY (id)
+    CONSTRAINT key_rotation_member_rotation_contact UNIQUE (rotation_id, contact_id),
+    CONSTRAINT key_rotation_member_rotation_contactgroup UNIQUE (rotation_id, contactgroup_id),
+    CONSTRAINT nonnulls_rotation_member_recipients_check CHECK (num_nonnulls(contact_id, contactgroup_id) = 1),
```

Comment:
That's not a good name choice for keeping the names consistent with MySQL/MariaDB as there's no such function there. Maybe something like `c(hec)k_one_of_contact_or_contactgroup` or `c(hec)k_one_recipient`.